website/content/docs/connect/gateways/api-gateway/configuration/http-route.mdx

```diff
@@ -0,0 +1,680 @@
+								}
+							]
+						}
+					],
+					"URLRewrite": [
+						{
+							"Path": "path to rewrite request to"
+						}
+					]
+				}
+			],
+			"Matches": [
+				{
+					"Headers": [
+						{
+							"Match": "type of match: exact, prefix or regex",
+							"Name": "name of header to match on",
+							"Value": "value of header to match on"
+						}
+					],
+					"Method": "method type to match on",
+					"Path": [
+						{
+							"Match": "type of match: exact, prefix or regex",
+							"Value": "value to match on"
+						}
+					],
+					"Query": [
+						{
+							"Match": "type of match: exact, present or regex",
+							"Name": "name of query parameter to match on",
+							"Value": "value of query parameter to match on"
+						}
+					]
+				}
+			],
+			"Services": [
+				{
+					"Filters": [
+						{
+							"Headers": [
+								{
+									"Add": [
+										{
+											"name of header to add": "value of header to add"
+										}
+									],
+									"Remove": ["header to remove from request"],
+									"Set": [
+										{
+											"name of header to set": "value of header to set"
+										}
+									]
+								}
+							],
+							"URLRewrite": [
+								{
+									"Path": "path to rewrite request to"
+								}
+							]
+						}
+					],
+					"Name": "name of Consul service to route to",
+					"Weight": "number proportional to other weights",
+          "Namespace": "enterprise: namespace of the route",
+          "Partition": "enterprise: partition of the route",
+				}
+			]
+		}
+	],
+	"Hostnames": [
+		"hostnames for which this HTTPRoute should respond to requests"
+	]
+}
+```
+
+</CodeTabs>
+
+## Specification
+
+This section provides details about the fields you can configure in the `http-route` configuration entry.
+
+### `Kind`
+
+Specifies the type of configuration entry to implement. For HTTP routes, this must be `http-route`.
+
+#### Values
+
+- Default: none
+- This field is required.
+- Data type: string value that must be set to `"http-route"`.
+
+### `Name`
+
+Specifies a name for the configuration entry. The name is metadata that you can
+use to reference the configuration entry when performing Consul operations,
+such as applying a configuration entry to a specific cluster.
+
+#### Values
+
+- Default: Defaults to the name of the node after writing the entry to the Consul server.
+- This field is required.
+- Data type: string
+
+### `Namespace` <EnterpriseAlert inline />
+
+Specifies the Enterprise [namespace](/consul/docs/enterprise/namespaces) to apply to the configuration entry.
+
+#### Values
+
+- Default: `"default"` in Enterprise
+- Data type: string
+
+### `Partition` <EnterpriseAlert inline />
+
+Specifies the Enterprise [admin partition](/consul/docs/enterprise/admin-partitions) to apply to the config entry.
+
+#### Values
+
+- Default: `"default"` in Enterprise
+- Data type: string
+
+### `Parents[]`
+
+Specifies the list of gateways that this route should be bound to.
```

Comment:
In this case I think we might want to leave "should be" or do something other than say, "binds" directly. This field does indeed describe the "ideal" state, as a route may fail to be bound to a gateway and you'll find out asynchronously on the route's status.

Reply:
maybe "will attempt to bind to"?

Reply:
Chatted w/ Andrew in Slack, and we think "ideal end state" is good for docs since that's what people will come in looking for, and then we tell them how to get there.